src/Url.php

```diff
@@ -1066,4 +1066,35 @@ public static function peelUrl($haystack, $prelude_string, $wrapper_start, $wrap
     }
     return FALSE;
   }
+
+  /**
+   * Checks if given URL matches a list of candidates for a default document.
+   *
+   * @param string $url
+   *   The URL to be tested
+   * @param array $candidates
+   *   A list of potential document names that could be indexes.
+   *   Defaults to "default" and "index".
+   *
+   * @return mixed
+   *   string - The base path if a matching document is found.
+   *   bool - FALSE if no matching document is found.
+   */
+  public static function getRedirectIfIndex($url, $candidates = array("default", "index")) {
+    // Assume no match
+    $root_path = FALSE;
+
+    // Filter through parse_url to separate out querystrings and etc.
+    $parsedUrl = parse_url($url);
+
+    // Test parsed URL
+    foreach ($candidates as $candidate) {
+      if(preg_match('/' . $candidate . '\.(htm(l?)|php/', $parsedUrl["path"])) {
+        $root_path = substr($parsedUrl, strlen($parsedUrl) - strlen($candidate));
+        break;
+      }
+    }
+
+    return $root_path;
```

Comment:
In most cases this will work.... but knowing that we have url address based migration coming in wireless, we will need this to not just discard query params and such.   You will need to re-attach those by just altering $parsed_url['path'] and then running $parsed_url back into self::reassembleURL($parsed_url, FALSE)

We need no babies getting thrown out with bathwater ;)